services/app-web/src/hooks/useContractForm.ts

```diff
@@ -0,0 +1,260 @@
+        } if (input.formData.supportingDocuments && input.formData.supportingDocuments.length > 0) {
+            input.formData.supportingDocuments = documentsInput(input.formData.supportingDocuments)
+        } if (input.formData.stateContacts && input.formData.stateContacts.length > 0) {
+            input.formData.stateContacts = stateContactsInput(input.formData.stateContacts)
+        }
+
+        try {
+            const updateResult = await updateFormData({
+                variables: {
+                    input: {
+                        contractID: contractID ?? 'new-draft',
+                        lastSeenUpdatedAt: contract!.draftRevision!.updatedAt,
+                        formData: input.formData
+                    },
+                },
+            })
+            const updatedSubmission =
+                updateResult?.data?.updateContractDraftRevision.contract
+            if (!updatedSubmission) {
+                setShowPageErrorMessage(true)
+                console.info('Failed to update form data', updateResult)
+                recordJSException(
+                    `StateSubmissionForm: Apollo error reported. Error message: Failed to update form data ${updateResult}`
+                )
+                return new Error('Failed to update form data')
+            }
+            return updatedSubmission
+        } catch (serverError) {
+            setShowPageErrorMessage(true)
+            recordJSException(
+                `StateSubmissionForm: Apollo error reported. Error message: ${serverError.message}`
+            )
+            return new Error(serverError)
+        }
+    }
+    const { 
+        data: fetchResultData,
+        error: fetchResultError,
+        loading: fetchResultLoading
+     } = useFetchContractQuery({
+        variables: {
+            input: {
+                contractID: contractID ?? 'new-draft'
+            }
+        },
+        skip: !contractID
+    })
+    const contract = fetchResultData?.fetchContract.contract
+    if (fetchResultLoading) {
+        interimState = 'LOADING'
+        return {interimState, createDraft, updateDraft, showPageErrorMessage }
+    }
+    if (fetchResultError) {
+        const err = fetchResultError
+        if (err instanceof ApolloError){
+            handleApolloError(err, true)
+            if (err.graphQLErrors[0]?.extensions?.code === 'NOT_FOUND') {
+                    interimState = 'NOT_FOUND'
+                    return {interimState, createDraft, updateDraft,  showPageErrorMessage }
+            }
+        }
+        if (err.name !== 'SKIPPED') {
+            recordJSException(err)
+            interimState = 'GENERIC_ERROR'// api failure or protobuf decode failure
+            return { interimState, createDraft, updateDraft,  showPageErrorMessage}
+        }
+
+        if (!contract || !contract.draftRevision || !contract.draftRevision.formData || contract?.status === 'RESUBMITTED' || contract?.status === 'SUBMITTED') {
```

Comment:
Now that we have UnlockedContract, I think this can be handled with a [type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html) like `isUnlockedContract`

Reply:
I was getting errors without this check iirc, but I will remove and see if the type guard works here